File: src/client/src/types/appointment.ts
export enum AppointmentStatus {
  Scheduled = 0,
  Confirmed = 1,
  InProgress = 2,
  Completed = 3,
  Cancelled = 4,
  NoShow = 5,
}

export enum AppointmentType {
  InPerson = 0,
  Phone = 1,
  VideoCall = 2,
  Emergency = 3,
}

export enum AppointmentPriority {
  Low = 0,
  Normal = 1,
  High = 2,
  Urgent = 3,
}

export interface AppointmentDetail {
  id: string;
  patientId: string;
  patientName: string;
  doctorId: string;
  doctorName: string;
  appointmentDate: string;
  startTime: string;
  endTime: string;
  status: AppointmentStatus;
  type: AppointmentType;
  reason?: string | null;
  notes?: string | null;
  priority: AppointmentPriority;
  durationMinutes: number;
  cancelledAt?: string | null;
  cancellationReason?: string | null;
  visitId?: string | null;
  createdAt: string;
  updatedAt?: string | null;
}

export interface AppointmentListItem {
  id: string;
  patientName: string;
  doctorName: string;
  appointmentDate: string;
  startTime: string;
  endTime: string;
  status: AppointmentStatus;
  type: AppointmentType;
  priority: AppointmentPriority;
}

export interface AppointmentCalendarItem {
  id: string;
  patientName: string;
  doctorName: string;
  appointmentDate: string;
  startTime: string;
  endTime: string;
  status: AppointmentStatus;
  type: AppointmentType;
}

export interface CreateAppointmentRequest {
  patientId: string;
  doctorId: string;
  appointmentDate: string;
  startTime: string;
  endTime: string;
  type: AppointmentType;
  reason?: string | null;
  notes?: string | null;
  priority: AppointmentPriority;
  durationMinutes: number;
}

export interface UpdateAppointmentRequest extends CreateAppointmentRequest {
  status: AppointmentStatus;
}

export interface CancelAppointmentRequest {
  cancellationReason: string;
}

export interface FollowUpItem {
  id: string;
  appointmentId: string;
  followUpDate: string;
  notes?: string | null;
  isCompleted: boolean;
  completedAt?: string | null;
  createdAt: string;
  updatedAt?: string | null;
}

export interface CreateFollowUpRequest {
  followUpDate: string;
  notes?: string | null;
}

export interface UpdateFollowUpRequest extends CreateFollowUpRequest {
  isCompleted: boolean;
}

File: src/Server/Infrastructure/Persistence/Entities/Appointment.cs
using Server.Core.Common;
using Server.Infrastructure.Persistence.Entities.Enums;

namespace Server.Infrastructure.Persistence.Entities;

public class Appointment : BaseEntity
{
    public Guid PatientId { get; set; }

    public Guid DoctorId { get; set; }

    public DateTime AppointmentDate { get; set; }

    public TimeSpan StartTime { get; set; }

    public TimeSpan EndTime { get; set; }

    public AppointmentStatus Status { get; set; } = AppointmentStatus.Scheduled;

    public AppointmentType Type { get; set; } = AppointmentType.InPerson;

    public string? Reason { get; set; }

    public string? Notes { get; set; }

    public AppointmentPriority Priority { get; set; } = AppointmentPriority.Normal;

    public int DurationMinutes { get; set; } = 30;

    public DateTime? CancelledAt { get; set; }

    public string? CancellationReason { get; set; }

    public Guid? VisitId { get; set; }

    public Patient? Patient { get; set; }

    public Doctor? Doctor { get; set; }

    public Visit? Visit { get; set; }

    public ICollection<FollowUp> FollowUps { get; set; } = new List<FollowUp>();
}
